### PR TITLE
Docker Tag command: Relax the restriction on namespace (username) length from 30 to 255 characters.

### PR DESCRIPTION
**Pull request by shishir-a412ed**
This change relaxes the restriction on namespace (username) from 30 to 255 characters.

Issue related to this: #10392

Shishir

### CODE REVIEW

File: registry/config.go [hunk 1]
```diff
@@ -223,7 +223,7 @@ func validateRemoteName(remoteName string) error {
 	if !validNamespaceChars.MatchString(namespace) {
 		return fmt.Errorf("Invalid namespace name (%s). Only [a-z0-9-_] are allowed.", namespace)
 	}
-	if len(namespace) < 4 || len(namespace) > 30 {
+	if len(namespace) < 4 || len(namespace) > 256 {
 		return fmt.Errorf("Invalid namespace name (%s). Cannot be fewer than 4 or more than 30 characters.", namespace)
```
Comment (ncdc) [inline registry/config.go]: You'll want to update the error text

Comment (shishir-a412ed) [inline registry/config.go]: Thanks @ncdc for the catch ! 
I have updated the code.

**Comment by thaJeztah** [comment]
Note to reviewers; it looks like the limit in registry v2 has not yet been updated, see https://github.com/docker/docker/issues/10392#issuecomment-76837217

File: registry/config.go [hunk 2]
```diff
@@ -223,8 +223,8 @@ func validateRemoteName(remoteName string) error {
 	if !validNamespaceChars.MatchString(namespace) {
 		return fmt.Errorf("Invalid namespace name (%s). Only [a-z0-9-_] are allowed.", namespace)
 	}
-	if len(namespace) < 4 || len(namespace) > 30 {
-		return fmt.Errorf("Invalid namespace name (%s). Cannot be fewer than 4 or more than 30 characters.", namespace)
+	if len(namespace) < 4 || len(namespace) > 256 {
```
Comment (ncdc) [inline registry/config.go]: The v2 registry currently has a minimum of 2, not 4: https://github.com/docker/distribution/blob/master/registry/api/v2/names.go#L16

**Comment by ncdc** [comment]
The current max in the v2 registry is 255, not 256: https://github.com/docker/distribution/blob/master/registry/api/v2/names.go#L32

**Comment by ncdc** [comment]
This PR for the v2 registry removes the 30-character component limit https://github.com/docker/distribution/pull/242

**Comment by crosbymichael** [comment]
@dmp42 @dmcgowan 

Please review

**Comment by dmp42** [comment]
@shishir-a412ed can you update the comparison to 255?
Thanks.

**Comment by dmcgowan** [comment]
Ping @ncdc can you drive forward getting this updated to match the rules in distribution

**Comment by ncdc** [comment]
Yes, I chatted with Shishir today about updating it to match. I'll follow
up on Monday. Thanks.

**Comment by stevvooe** [comment]
Updated ticket title to reflect the correct limit.

File: registry/registry_test.go [hunk 3]
```diff
@@ -776,9 +779,6 @@ func TestValidRemoteName(t *testing.T) {
 		// Disallow consecutive hyphens.
 		"dock--er/docker",
 
-		// Namespace too short.
```
Comment (stevvooe) [inline registry/registry_test.go]: Could replace this with a testcase that triggers the short case?

**Comment by stevvooe** [comment]
LGTM with testcase for short namespace.

**Comment by dmcgowan** [comment]
Commit message hasn't been changed to reflect 255 instead of 256 characters

**Comment by shishir-a412ed** [comment]
@dmcgowan Updated the commit message to reflect 255.

**Comment by shishir-a412ed** [comment]
@stevvooe I have added 2 testcases, one for short, and one for long namespaces.

**Comment by dmcgowan** [comment]
LGTM

**Comment by stevvooe** [comment]
LGTM

**Comment by ncdc** [comment]
@icecrime what are the next steps here?

**Comment by vbatts** [comment]
i've just issued a rebuild to windows, but the linux tests are looking fine.

**Comment by icecrime** [comment]
Windows CI is green after rebuild: pushing to `3-docs-review`.

**Comment by ncdc** [comment]
@icecrime thanks. FYI I grepped through the Docker codebase and the only references I could find (quickly) to a 30 character maximum were the docker io API and the hub registry spec docs, both of which talk about hub/registry usernames needing to be no more than 30.

**Comment by icecrime** [comment]
Good, I was running the same search on my side and can't see any reference to these limits in `docs/` either :-)

**Pull request merged**